rootston/output.c

```diff
@@ -133,23 +136,52 @@ static void render_view(struct roots_view *view, struct roots_desktop *desktop,
 	}
 }
 
+struct wl_list *render_layer_surfaces(struct wl_list *layer_surfaces,
+		struct roots_desktop *desktop, struct wlr_output *wlr_output,
+		struct timespec *when, enum surface_layers_layer until_layer) {
```

Comment:
I think it would be a whole lot cleaner to have a function that renders surfaces by layer so you don't have to do math on the enum or return anything. But I'm probably doing a big rootston refactor pretty soon where I'll clean up a lot of stuff.

Reply:
That's what I did the first time, and that caused a lot of issues, in particular when managing pointer focus. Using the sorted list is cool because it allows to iterate through the list of layer surfaces only once.

Reply:
We can make this function only render one layer if we want to, though. Alternatively, maybe a `wlr_surface_layers_layer_cmp` function might be useful. What do you think?

Reply:
> Using the sorted list is cool because it allows to iterate through the list of layer surfaces only once.

If you're trying to optimize the code, please spend your time on the renderer instead :).

Yeah, I think rendering one layer at a time would help readability of this section quite a bit.

Reply:
I don't think rendering one layer at a time helps readability. Also, this isn't the most complex function - there's another one in `cursor.c` that is harder to understand, and also returns a list of remaining layer surfaces. Ironically, I think making these functions more "simple" would complexify them.